eng/packages/http-client-csharp/generator/Azure.Generator/test/TestHelpers/MockHelpers.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.SourceInput;
+using Moq;
+using Moq.Protected;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Azure.Generator.Tests.TestHelpers
+{
+    internal class MockHelpers
+    {
+        private static readonly string _configFilePath = Path.Combine(AppContext.BaseDirectory, TestHelpersFolder);
+        public const string TestHelpersFolder = "TestHelpers";
+
+        public static Mock<AzureClientPlugin> LoadMockPlugin(
+            Func<InputType, TypeProvider, IReadOnlyList<TypeProvider>>? createSerializationsCore = null,
+            Func<InputType, CSharpType>? createCSharpTypeCore = null,
+            Func<InputApiKeyAuth>? apiKeyAuth = null,
+            Func<IReadOnlyList<string>>? apiVersions = null,
+            Func<IReadOnlyList<InputEnumType>>? inputEnums = null,
+            Func<IReadOnlyList<InputModelType>>? inputModels = null,
+            Func<IReadOnlyList<InputClient>>? clients = null)
+        {
+            IReadOnlyList<string> inputNsApiVersions = apiVersions?.Invoke() ?? [];
+            IReadOnlyList<InputEnumType> inputNsEnums = inputEnums?.Invoke() ?? [];
+            IReadOnlyList<InputClient> inputNsClients = clients?.Invoke() ?? [];
+            IReadOnlyList<InputModelType> inputNsModels = inputModels?.Invoke() ?? [];
+            InputAuth inputNsAuth = apiKeyAuth != null ? new InputAuth(apiKeyAuth(), null) : new InputAuth();
+            var mockTypeFactory = new Mock<AzureTypeFactory>() { CallBase = true };
+            var mockInputNs = new Mock<InputNamespace>(
+                string.Empty,
+                inputNsApiVersions,
+                inputNsEnums,
+                inputNsModels,
+                inputNsClients,
+                inputNsAuth);
+            var mockInputLibrary = new Mock<InputLibrary>(_configFilePath);
+            mockInputLibrary.Setup(p => p.InputNamespace).Returns(mockInputNs.Object);
+
+            if (createCSharpTypeCore is not null)
+            {
+                mockTypeFactory.Protected().Setup<CSharpType>("CreateCSharpTypeCore", ItExpr.IsAny<InputType>()).Returns(createCSharpTypeCore);
+            }
+
+            // initialize the mock singleton instance of the plugin
+            var codeModelInstance = typeof(AzureClientPlugin).GetField("_instance", BindingFlags.Static | BindingFlags.NonPublic);
+            var clientModelInstance = typeof(AzureClientPlugin).GetField("_instance", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
these two are the same, why do we need 2 instances of them?

Reply:
Oh - I did it wrong. Let me fix.